src/User/Hander/UserRegistrationHandler.php

```
@@ -41,5 +41,7 @@ public function handle(RegisterUser $command)
         );
 
         $this->userRepository->save($user);
```

Quick question - if repository doesn't persist (for whatever reason), will it fail hard (throw an exception) or just emit some kind of warning/return false?

If it doesn't fail hard, next line will publish an event that's just false.

If it does fail hard, the exception will bubble up to some upper layer of app.

Should we:
1. wrap this in a try/catch block and publish some other kind of event here
2. let exception bubble up to upper layers and handle them there (publish an app/infra lvl event or just handle it some other way)

Is the above explained case an application level or infrastructure level exception?

---

Repository should throw an exception, it would bubble up to top.

Exception would most likely be infrastructural